@heaths @jsquire I added the targets to `Azure.Core.Tests.csproj` locally to try and run the test suite. I got the following error when building:

> C:\...Azure.Core.Tests.csproj : error NU1701: Warning As Error: Package 'NUnit3TestAdapter 4.4.2' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETCoreApp,Version=v2.1'. This package may not be fully compatible with your project.

I hid that one and then got the following:
> C:\Users\mredding\.nuget\packages\system.runtime.compilerservices.unsafe\6.0.0\buildTransitive\netcoreapp2.0\System.Runtime.CompilerServices.Unsafe.targets(4,5): error : System.Runtime.CompilerServices.Unsafe doesn't support netcoreapp2.1. Consider updating your TargetFramework to netcoreapp3.1 or later. [C:\Users\mredding\source\repos\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj::TargetFramework=netcoreapp2.1]

Which I'm 90% sure is coming from having removed: 
```
  <!-- Keep System.Diagnostics.DiagnosticSource 4.6.0 for netcoreapp2.1 support. -->
  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.1' ">
    <PackageReference Include="System.Diagnostics.DiagnosticSource" VersionOverride="4.6.0" />
  </ItemGroup>
```
Because that was the error that was causing me to need to keep that additional version in the first place. **I think this means that with this change we would be breaking customers using netcoreapp2.1**.

I removed the netcoreapp2.1 target and ran the Azure.Core.Tests for net5.0, which had three failures:

This one for both async/sync, with the failure: `System.InvalidCastException : Unable to cast object of type 'System.Net.Http.HttpClientHandler' to type 'System.Net.Http.SocketsHttpHandler'.`
```csharp
#if NETCOREAPP
        [Test]
        public void UsesSocketsClientTransportOnNetCoreApp()
        {
            var transport = (HttpClientTransport)GetTransport();
            HttpClient transportClient = transport.Client;
            var handler = (SocketsHttpHandler)GetHandler(transportClient);
            Assert.AreEqual(TimeSpan.FromSeconds(300), handler.PooledConnectionLifetime);
            Assert.GreaterOrEqual(handler.MaxConnectionsPerServer, 50);
        }
#else
        [Test]
        public void UsesHttpClientTransportOnNetStandart()
        {
            var transport = (HttpClientTransport)GetTransport();
            HttpClient transportClient = transport.Client;
            var handler = (HttpClientHandler)GetHandler(transportClient);
            Assert.AreEqual(50, handler.MaxConnectionsPerServer);
        }
#endif
```
And this one with `Expected: True But was: False`
```csharp
        public void CreateOperationCanceledExceptionPassesTokenOnSupportedVersions()
        {
            using var cts = new CancellationTokenSource();
            cts.Cancel();
            var ex = (TaskCanceledException) CancellationHelper.CreateOperationCanceledException(new Exception("test"), cts.Token);

#if NETCOREAPP2_1_OR_GREATER
            Assert.IsTrue(ex.CancellationToken.IsCancellationRequested);
#else
            Assert.IsFalse(ex.CancellationToken.IsCancellationRequested);
#endif
```
I think both of these tests seem to be a case of the test just not picking up on the correct target, since these tests haven't been run with a net5.0 target in... awhile? It doesn't seem like the behavior is incorrect since the outcome fits the else cases.

Do either of you have thoughts about if this impacts the work in this PR?

--- a/sdk/core/Azure.Core/CHANGELOG.md
+++ b/sdk/core/Azure.Core/CHANGELOG.md
@@ -9,7 +9,7 @@
 ### Bugs Fixed
 
 ### Other Changes
-- Remove targets for .NET Core 2.1 and .NET 5 since they are out of support. Remaining targets are unchanged.
+- Remove targets for .NET Core 2.1 and .NET 5 since they are out of support. Azure.Core is no longer compatible with .NET Core 2.1 after removal of target. The remaining targets are unchanged.
 
 ## 1.36.0 (2023-11-10)
 

